### DISCUSSION TITLE
How can I get the minimum distance between two objects along a specific direction?

### DISCUSSION THREAD

**Discussion by Terryxyz** [shapely/shapely#1317] [Q&A]
For example, if we have two objects (for example, polygon A and polygon B), the question is, if polygon A moves in a specific direction (for example, direction [1,0]), after moving how much distance will it collide with polygon B?

**Comment by Jeremiah-England** [comment]
I made a gist that does it [here](https://gist.github.com/Jeremiah-England/6377b798d6384d454583e55fc053f3cc). I think this question, at least with how it's worded, classifies as a 'usage question' which are better suited to the [gis.stackexchange](https://gis.stackexchange.com) site. But I enjoyed working on it, so thanks for asking. I could also see it being worded as a feature request, though I don't have plans to try to incorporate code in the gist into shapely. I'm not sure what makes a feature belong in shapely.

If you still need a solution to this problem and have any questions about the gist, feel free to ask them in a comment there. I'd be happy to help and be glad to have a second opinion.
